src/Omega_h_mesh2d.cpp

```diff
@@ -12,7 +12,7 @@ void Mesh2D::set_dim(Int dim_in) {
   dim_ = dim_in;
 }
 
-std::optional<Model2D> Mesh2D::getModel() const {
+std::optional<const Model2D> Mesh2D::getModel() const {
```

Comment:
This is probably not what you want. I'm pretty sure this will make a copy of the model to return since `std::optional` owns the data. A pointer is probably the way to go here, but with an added note in the docs that it may be null since our users may not be as careful with checking for `nullptr` as they should be.

If you return a pointer, you can also mark this function noexcept